lib/store/storage.js

```diff
@@ -194,15 +237,99 @@ var uploadFileIfNotExists = module.exports.uploadFileIfNotExists = function(file
         log.info('Possible bad request or S3 params failure. Check if S3 multi part uploads or transfer acceleration is enabled');
         return callback(err);
 
-      } else {
-        // TODO: Error handling will be further improved once metadata collection process is added.
-        return callback(err);
       }
 
+      // TODO: Error handling will be further improved once metadata collection process is added.
+      return callback(err);
+
     } else if (data) {
       log.info('File already exists. Skipping');
       return callback(null, 'File already exists. Skipping');
 
     }
   });
 };
+
+/**
+ * Based on the parameters provided list all the objects within a Bucket Path
+ * This step is a pre-cursor to comparing the list of objects with latest sync files.
+ * If any of the files are out of sync after refresh they will be deleted
+ *
+ * @param  {String}           filename               Name of the file downloaded from cnavas data
+ * @param  {Function}         callback               Standard callback function
+ * @param  {Object}           callback.err           An error object, if any
+ */
+var cleanUpobsoleteFiles = module.exports.cleanUpobsoleteFiles = function(latestSyncFileNames, callback) {
+  // Get prefixed objects
+  var bucket = config.get('datalake.s3.bucket');
+  var path = util.format('%s/%s/',
+    config.get('datalake.canvasData.directory.currentTerm'),
+    'requests'
+  );
+  var obsoleteFiles = [];
+
+  listAllObjects({bucket: bucket, prefix: path}, function(err, data) {
+    if (err) {
+      log.error('Could not list all the objects in the bucket.');
+      return callback(err);
+
+    }
+
+    log.info('Got ' + data.length + ' objects with prefix.');
+    async.eachSeries(data, function(s3File, done) {
+      var s3FileName = s3File.Key.split(path)[1];
+      var s3Key = s3File.Key;
```

Comment:
@sandeepmjay Are the oddly capitalized property names here and below a requirement of the AWS client library?

Reply:
Yes. AWS folks didn't have a good linter.  The properties are obtained from the S3's list all bucket objects API feeds.